flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/collect/CollectSinkOperatorCoordinatorTest.java

```diff
@@ -105,54 +102,147 @@ void testCloseAfterRequestIsReceived() throws Exception {
+            assertThat(responseFuture).isNotDone();
+
+            socketServer.handleRequest(expectedData);
+
+            assertResponseWithDefaultMetadataFromServer(responseFuture, expectedData);
+        }
+    }
+
+    @Test
+    void testServerSideClosingTheServerSocket() throws Exception {
+        try (CollectSinkOperatorCoordinator coordinator = new CollectSinkOperatorCoordinator()) {
+            coordinator.start();
+
+            final CompletableFuture<CoordinationResponse> responseFuture;
+            try (final TestingSocketServer socketServer =
+                    TestingSocketServer.createSocketServerAndInitializeCoordinator(coordinator)) {
+                final String version = "version";
+                responseFuture =
+                        coordinator.handleCoordinationRequest(
+                                createRequestForClientGeneratedResponse(version));
+                assertThat(responseFuture).isNotDone();
+
+                FlinkAssertions.assertThatFuture(socketServer.handleRequestWithoutResponse())
+                        .eventuallySucceeds();
+            }
+            assertEmptyResponseGeneratedFromServer(responseFuture);
+        }
+    }
+
+    @Test
+    void testServerSideClosingTheAcceptingSocket() throws Exception {
+        try (CollectSinkOperatorCoordinator coordinator = new CollectSinkOperatorCoordinator()) {
+            coordinator.start();
+
+            try (final TestingSocketServer socketServer =
+                    TestingSocketServer.createSocketServerAndInitializeCoordinator(coordinator)) {
+                final String version = "version";
+                final CompletableFuture<CoordinationResponse> responseFuture =
+                        coordinator.handleCoordinationRequest(
+                                createRequestForClientGeneratedResponse(version));
+                assertThat(responseFuture).isNotDone();
+
+                final CompletableFuture<List<Row>> dataFuture = new CompletableFuture<>();
+                final CompletableFuture<Void> serverSideProcess =
+                        socketServer.handleRequestAsync(dataFuture);
+                assertThat(serverSideProcess).isNotDone();
+
+                // wait for the connection to be established
+                socketServer.waitForConnectionToBeEstablished();
+                // in order to close the connection from the server's side
+                socketServer.closeAcceptingSocket();
+
+                assertEmptyResponseGeneratedFromClient(responseFuture, version);
+
+                assertThat(serverSideProcess).isNotDone();
 
-            // server closes here
-            failedResponseFuture =
+                dataFuture.complete(Collections.singletonList(Row.of(123, "abc")));
+
+                FlinkAssertions.assertThatFuture(serverSideProcess)
+                        .eventuallyFailsWith(ExecutionException.class)
+                        .withCauseInstanceOf(SocketException.class);
```

Comment:
This seems to primarily be testing the `TestingSocketServer`.

Reply:
I would completely drop the last 2 assertion blocks

Reply:
I also removed the lines above (see [comment](https://github.com/apache/flink/pull/23296#discussion_r1308284339)). Injecting the data is also not necessary in `testClosingTheListeningSocketInTheSinkFunction` (aka `testServerSideClosingTheAcceptingSocket`)